### PR TITLE
Update test run generator for brave-core

### LINKED ISSUE #93: Update test run generator script to match brave-core

### PR DESCRIPTION
**Pull request by srirambv**
Fixes #93

Milestones needs to organized on repo and that needs updating in future once we get to proper channel releases. This is only for temp changes to add issues with `QA/Needed` labels to test sheet.

### CODE REVIEW

**Comment by kjozwiak** [comment]
@srirambv there seems to be a bug here:

https://github.com/brave/qa-resources/blob/4bb2dcfd0fd90c2a0df076cf13af98d7016eb9ce/brave_testrun_generator.py#L158.

It also includes `or 'release-notes/exclude' in label_names` which will basically add anything that's labelled as 'release-notes/exclude' into `release_notes.append(output_line)`.

Example with `or 'release-notes/exclude' in label_names`:

```
Create test runs for: 1

Generating Per-release checklist for 0.56.x - Release Hotfix 1 on all platforms
0.56.x - Release Hotfix 1
Release Notes
 - Cookies not imported - follow up to 1545.([#2013](https://github.com/brave/brave-browser/issues/2013))
 - Stats not imported during auto-upgrade.([#2014](https://github.com/brave/brave-browser/issues/2014))
 - Upgrade from Chromium 70.0.3538.77 to 70.0.3538.102.([#2099](https://github.com/brave/brave-browser/issues/2099))
```

Example without `or 'release-notes/exclude' in label_names`:

```
Create test runs for: 1

Generating Per-release checklist for 0.56.x - Release Hotfix 1 on all platforms
0.56.x - Release Hotfix 1
Release Notes
 - Upgrade from Chromium 70.0.3538.77 to 70.0.3538.102.([#2099](https://github.com/brave/brave-browser/issues/2099))
```

File: wikitemplate.md [hunk 1]
```diff
@@ -1,282 +1,170 @@
...
+### Installer
 
-## Last changeset test
+- [ ] Check that installer is close to the size of last release
+- [ ] Check signature: If OS Run `spctl --assess --verbose /Applications/Brave-Browser-Beta.app/` and make sure it returns `accepted`.  If Windows right click on the `brave_installer-x64.exe` and go to Properties, go to the Digital Signatures tab and double click on the signature.  Make sure it says "The digital signature is OK" in the popup window
 
-- [ ] Test what is covered by the last changeset (you can find this by clicking on the SHA in about:brave)
 
-## Data
+### Data
```
[the remark this answers is not in the document]
Comment (srirambv) [reply wikitemplate.md]: Will do a follow up PR for that

**Comment by srirambv** [comment]
Fixed review comments for template and iOS. I'll do a follow up PR for the release notes. Will take it all out since Changelog generator is created for that now.

File: wikitemplate.md [hunk 2]
```diff
@@ -1,282 +1,170 @@
...
+- [ ] Verify that `chrome://` forwards to `brave://`
+- [ ] Verify `brave://adblock` loads adblock page
+- [ ] Verify `brave://rewards` loads Brave rewards page 
+- [ ] Verify `brave://newtab` loads a new tab
+- [ ] Verify `brave://welcome` loads the welcome page 
+- [ ] Verify `brave://version` correctly shows Brave version and Chromium version 
+
+### Import tests
+
+- [ ] Verify that you can import bookmarks, cookies, history, passwords and stats from Brave(browser-laptop)
```
[the remark this answers is not in the document]
Comment (srirambv) [reply wikitemplate.md]: Removed and added as muon

File: wikitemplate.md [hunk 3]
```diff
@@ -1,282 +1,170 @@
...
+- [ ] Verify that you can import bookmarks from Edge
+- [ ] Verify importing `Bookmark HTML` file retains the folder structure on a clean profile
+
+### Context menus
+
+- [ ] Verify `Block element via selector` removes a CSS element from page without reloading
+- [ ] Verify `Clear all CSS rules for this sites` removes the blocked element after page reload
+- [ ] Verify `Clear all CSS rules for all sites` removes the rule and loads all elements for all pages
+
+### Keyboard Shortcuts (WIP)
```
[the remark this answers is not in the document]
Comment (srirambv) [reply wikitemplate.md]: fixed

File: wikitemplate.md [hunk 4]
```diff
@@ -1,282 +1,170 @@
...
+- [ ] Visit https://browserleaks.com/geo in a Tor window, ensure location isn't shown
+- [ ] Visit https://diafygi.github.io/webrtc-ips/ in a Tor window with block all fingerprinting, ensure WebRTC is blocked and no IP is shown
+- [ ] Visit https://diafygi.github.io/webrtc-ips/ in a Tor window, disable shields, ensure WebRTC is blocked and no IP is shown
+- [ ] Verify Flash doesn't work on Tor window even if it is enabled in `brave://settings/content/flash`
+- [ ] Verify Torrent viewer doesn't load in a Tor window and warns when trying to load a torrent/magnet link in a Tor window
+- [ ] Verify Google Widevine doesn't load in Tor window and doesn't prompt Google Windevine notification in URL bar
+- [ ] Ensure you are able to download a file in Tor window. Verify all Download/Cancel, Download/Retry and Download works in Tor window
+- [ ] Disconnect network and open a Tor window, should show modal to retry connection or relaunch Tor window to connect 
+
+### Sync (WIP)
```
[the remark this answers is not in the document]
Comment (srirambv) [reply wikitemplate.md]: fixed

File: wikitemplate.md [hunk 5]
```diff
@@ -1,282 +1,170 @@
...
+
+### Rewards (WIP)
+
+####Rewards Media (To be verified on YouTube and Twitch) (WIP)
```
[the remark this answers is not in the document]
Comment (srirambv) [reply wikitemplate.md]: Fixed

File: wikitemplate.md [hunk 6]
```diff
@@ -1,282 +1,170 @@
...
+### Crash Reporting
 
-- [ ] Check that loading `chrome://crash` causes the new tab to crash 
-- [ ] Check that `chrome://crashes` lists all the crashes and includes both Crash Report ID & Local Crash ID
+- [ ] Check that loading `brave://crash` causes the new tab to crash
+- [ ] Check that `brave://crashes` lists all the crashes and includes both Crash Report ID & Local Crash ID
 - [ ] Verify the crash ID matches the report on brave stats
 
-## Performance test
+### Performance test
```
[the remark this answers is not in the document]
Comment (srirambv) [reply wikitemplate.md]: Fixed removed sync related. Will add it back once its implemented fully

File: wikitemplate.md [hunk 7]
```diff
@@ -1,282 +1,170 @@
...
 
 ## Extensions/Plugins tests
 
-- [ ] Enable each extension one by one under `about:preferences#extensions` and ensure that the browser doesn't become unresponsive
-- [ ] Enable 1Password and ensure that the basic functionality is working
-- [ ] Enable LastPass and ensure that the basic functionality is working
-- [ ] Enable Dashlane and ensure that the basic functionality is working
-- [ ] Enable bitwarden and ensure that the basic functionality is working
+- [ ] Verify one item from Brave Update server is installable (Example: Ad-block DAT file on fresh extension)
+- [ ] Verify one item from Google Update server is installable(Example: Extensions from CWS)
```
[the remark this answers is not in the document]
Comment (srirambv) [reply wikitemplate.md]: Fixed

File: wikitemplate.md [hunk 8]
```diff
@@ -1,282 +1,170 @@
...
+- [ ] Verify older version of an extension gets updated to new version via Brave server 
+- [ ] Verify magnet links and .torrent files loads Torrent viewer page and able to download torrent
 
-### Widevine
-  - [ ] Test that you can log into Netflix and start a show
+### CWS
+
+- [ ] Verify installing ABP from CWS shows warning message `NOT A RECOMMENDED BRAVE EXTESION!`but still allows to install the extension
+- [ ] Verify installing LastPass from CWS doesn't show any warning message 
+- [ ] Verify installing an extesion that is not vetted by Brave gets blocked
```
[the remark this answers is not in the document]
Comment (srirambv) [reply wikitemplate.md]: fixed

File: wikitemplate.md [hunk 9]
```diff
@@ -1,282 +1,170 @@
...
+- [ ] Verify PDFJS, Torrent viewer extensions are installed automatically on fresh profile and cannot be disabled
+- [ ] Verify older version of an extension gets updated to new version via Google server
+- [ ] Verify older version of an extension gets updated to new version via Brave server 
+- [ ] Verify magnet links and .torrent files loads Torrent viewer page and able to download torrent
 
-### Widevine
-  - [ ] Test that you can log into Netflix and start a show
+### CWS
+
+- [ ] Verify installing ABP from CWS shows warning message `NOT A RECOMMENDED BRAVE EXTESION!`but still allows to install the extension
```
[the remark this answers is not in the document]
Comment (srirambv) [reply wikitemplate.md]: fixed

File: wikitemplate.md [hunk 10]
```diff
@@ -1,282 +1,170 @@
...
 - [ ] Turning HTTPS Everywhere off and shields off both disable the redirect to https://https-everywhere.badssl.com/
-- [ ] Check that ad replacement works on http://slashdot.org
-- [ ] Check that toggling to blocking and allow ads works as expected
-- [ ] Test that clicking through a cert error in https://badssl.com/ works
-- [ ] Visit several popular websites and ensure that the certificate viewer is working correctly (macOS/Windows only)
-  - [ ] Ignore the certificate warning under https://expired.badssl.com and ensure that the certificate viewer is working correctly once the page loads
-- [ ] Turning Safe Browsing off and shields off both disable safe browsing for https://www.raisegame.com/
-- [ ] Visit https://brianbondy.com/ and then turn on script blocking, nothing should load. Allow it from the script blocking UI in the URL bar and it should work
-- [ ] Test that about:preferences default Bravery settings take effect on pages with no site settings
+- [ ] Verify that toggling `Ad Control` works as expected
```
[the remark this answers is not in the document]
Comment (srirambv) [reply wikitemplate.md]: fixed

File: wikitemplate.md [hunk 11]
```diff
@@ -1,282 +1,170 @@
...
+- [ ] Visit https://protonirockerxow.onion/ in a Tor window, ensure login page is shown
+- [ ] Visit http://pdf995.com in a Tor window, should warn about connecting to HTTP site
+- [ ] Visit https://browserleaks.com/geo in a Tor window, ensure location isn't shown
+- [ ] Visit https://diafygi.github.io/webrtc-ips/ in a Tor window with block all fingerprinting, ensure WebRTC is blocked and no IP is shown
+- [ ] Visit https://diafygi.github.io/webrtc-ips/ in a Tor window, disable shields, ensure WebRTC is blocked and no IP is shown
+- [ ] Verify Flash doesn't work on Tor window even if it is enabled in `brave://settings/content/flash`
+- [ ] Verify Torrent viewer doesn't load in a Tor window and warns when trying to load a torrent/magnet link in a Tor window
+- [ ] Verify Google Widevine doesn't load in Tor window and doesn't prompt Google Windevine notification in URL bar
+- [ ] Ensure you are able to download a file in Tor window. Verify all Download/Cancel, Download/Retry and Download works in Tor window
+- [ ] Disconnect network and open a Tor window, should show modal to retry connection or relaunch Tor window to connect
```
[the remark this answers is not in the document]
Comment (srirambv) [reply wikitemplate.md]: Have retained for now. Will remove in the follow up pr for removing release section

File: wikitemplate.md [hunk 12]
```diff
@@ -1,282 +1,170 @@
...
-- [ ] Test that using `BRAVE_ENABLE_PREVIEW_UPDATES=TRUE` env variable works and prompts for preview build updates.
+- [ ] Verify visiting `brave://settings/help` triggers update check
+- [ ] Verify once update is downloaded, prompts to `Relaunch` to install update
+
+
+## Chromium upgrade tests
+
+#### Adblock
+
+- [ ] Verify referrer blocking works properly for TLD+1
```
[the remark this answers is not in the document]
Comment (srirambv) [reply wikitemplate.md]: provided a verification check

File: wikitemplate.md [hunk 13]
```diff
@@ -1,282 +1,170 @@
...
+### Sync (WIP)
+
+### Rewards (WIP)
+
+####Rewards Media (To be verified on YouTube and Twitch) (WIP)
+
+### Cookie and Cache
 
-## Cookie and Cache
+- [ ] Go to http://samy.pl/evercookie/ and set an evercookie. Check that going to prefs, clearing site data and cache, and going back to the Evercookie site does not remember the old evercookie value
```
[the remark this answers is not in the document]
Comment (srirambv) [reply wikitemplate.md]: fixed

File: wikitemplate.md [hunk 14]
```diff
@@ -1,282 +1,170 @@
...
+- [ ] Visit https://diafygi.github.io/webrtc-ips/ in a Tor window, disable shields, ensure WebRTC is blocked and no IP is shown
+- [ ] Verify Flash doesn't work on Tor window even if it is enabled in `brave://settings/content/flash`
+- [ ] Verify Torrent viewer doesn't load in a Tor window and warns when trying to load a torrent/magnet link in a Tor window
+- [ ] Verify Google Widevine doesn't load in Tor window and doesn't prompt Google Windevine notification in URL bar
+- [ ] Ensure you are able to download a file in Tor window. Verify all Download/Cancel, Download/Retry and Download works in Tor window
+- [ ] Disconnect network and open a Tor window, should show modal to retry connection or relaunch Tor window to connect 
+
+### Sync (WIP)
+
+### Rewards (WIP)
```
[the remark this answers is not in the document]
Comment (srirambv) [reply wikitemplate.md]: fixed

**Pull request merged**